R/chop.R

```diff
@@ -90,188 +90,253 @@ chop <- function(data, cols) {
 #' @export
 #' @rdname chop
 unchop <- function(data, cols, keep_empty = FALSE, ptype = NULL) {
-  cols <- tidyselect::eval_select(enquo(cols), data)
-  if (length(cols) == 0) {
-    return(data)
-  }
+  sel <- tidyselect::eval_select(enquo(cols), data)
 
-  if (keep_empty) {
-    for (col in cols) {
-      data[[col]][] <- map(data[[col]], init_col)
-    }
+  if (is_null(ptype)) {
+    # Convert to form that `df_unchop()` requires
+    ptype <- list()
   }
 
-  # In case `x` is a grouped data frame and any `cols` are lists,
+  size <- vec_size(data)
+  names <- names(data)
+
+  # In case `data` is a grouped data frame and any `cols` are lists,
   # in which case `[.grouped_df` will error
-  cols <- new_data_frame(unclass(data)[cols])
+  out <- new_data_frame(data, n = size)
+  cols <- out[sel]
+
+  # Remove unchopped columns to avoid slicing them needlessly later
+  out[sel] <- NULL
```

Comment:
@mgirlich had the nice idea that we should remove the columns that get unchopped up front. That way when we do `vec_slice(out, loc)` later on, we don't needlessly slice those columns (since we are going to immediately replace them with `update_cols()`)